### ISSUE TITLE
Editing after capture does not remove files referenced only by deleted elements

### ISSUE THREAD

**Issue by iiv3** [danny0838/webscrapbook#351]
Capture a page with images.
Open the captured page from scrapbook sidebar.
Use "DOM Eraser" from WSB toolbar to remove (some of) the images.
Use "Save" from the WSB toolbar.

Even if the "index.html" no longer references the images, the image files remain in place. It happens with "folder" and "htz" with backend server.

If editing is done before capture, the erased image files are never stored.

However it is not always possible to remove all unwanted elements on live website who have JS code that constantly loads more and more stuff.

---
The original Scrapbook used to remove files that were no longer linked/used, so I kind of expected the same behavior.

You might consider this feature request.

If my observations are correct, on "Save" changes are done in place, so leaving old stuff is just side effect"

I would consider it an improvement, if on "Save" a new archive is created and the old one is moved to "recycle bin". (Might be good idea to do that at recapture too.)

**Comment by danny0838** [comment]
No, legacy ScrapBook doesn't support this feature. There has been a [similar request](https://github.com/danny0838/firefox-scrapbook/issues/205), too.

It's not easy to scan all unused resources in an item, especially when taking account of JavaScript related content, CSS images, shadow root, erased content (which is revertible unless explicitly deleted), etc. We probably cannot implement it in near future.

**Comment by iiv3** [comment]
You don't need to scan for removed content.

You just make a new capture and add the content that is referenced. You just don't run all the conversion scripts and preserve the original meta data.

It's possible to do a new "capture page as..." of the captured and edited page, but you lose e.g. the "Source" URL field. No idea what happens with the html, but it also contains some metadata.

---
BTW, I wasn't talking about "Scrapbook X".
I was talking for the original "Scrapbook". It definitely removed unused files.

**Comment by danny0838** [comment]
> You don't need to scan for removed content.
> 
> You just make a new capture and add the content that is referenced. You just don't run all the conversion scripts and preserve the original meta data.
> 

This won't be any easier, "even preserve metadata", and "replace all original item files" will require a large code work.

You can try implementing it to see if its true.

> It's possible to do a new "capture page as..." of the captured and edited page, but you lose e.g. the "Source" URL field. No idea what happens with the html, but it also contains some metadata.
> 
> BTW, I wasn't talking about "Scrapbook X". I was talking for the original "Scrapbook". It definitely removed unused files.

ScrapBook X is derived from ScrapBook nd we've been porting new features from ScrapBook. It's certain that ScrapBook doesn't have such feature. It you think I'm wrong, provide the exact ScrapBook version for further investigation.
